### ISSUE TITLE
Resource Not Found Error

### ISSUE THREAD

**Issue by papa28x4** [cloudinary/cloudinary_php#280]
Hi, thanks for the good work.

Currently trying out Cloudinary and I have an issue.

I upload using:
```
 $result = $cloudinary->uploadApi()->upload($file->getRealPath(), [ 
            "resource_type" => "video",
            "folder" => "store",
            "type"=> "private"
        ]);
```
I get a success response. I can also see the video in cloudinary.

Then I try to download using:
`$downloadUrl = (new UploadApi)->privateDownloadUrl("store/tour", "mp4", array("attachment" => true));`

a download link is created

But when I attempt to use the link I get this:

```
{
    error: {
    message: "Resource not found - store/tour"
    }
}

```
Please what am I doing wrong?

**Comment by leptians** [comment]
@papa28x4 , `type: private` is only to protect the original and required a signed URL to access it while allowing access to its derived. You can refer to the following [documentation](https://cloudinary.com/documentation/control_access_to_media#private_media_assets) for more info.
In the mean time, `privateDownloadUrl` is a mean to generate private archive.
If what you are interested in just accessing the uploaded `private` asset, you can use the following to generate the signed URL:

```
echo $cloudinary->image($public_id)->private_($public_id)->signUrl();
```

Hope that helps

**Comment by papa28x4** [comment]
Thanks for the reply @leptians 

I want to provide time-limited access to private media assets.

This is scenario I have in mind.  A buyer purchases a photo, and I want to deliver the original image to him (and only to him) while our originals are inaccessible to anyone else. I need to provide him a download link he can use for a limited time.

will this `$cloudinary->image($public_id)->private_($public_id)->signUrl();` suffice

**Comment by papa28x4** [comment]
**Update**

The earlier code works fine with images. The challenge is with videos. A deeper look into it made discover the privateDownloadUrl method does not take into account the resource_type and will always generate this base url [](https://api.cloudinary.com/v1_1/papa28x4/image/download) instead of [](https://api.cloudinary.com/v1_1/papa28x4/video/download) for non-image media files. Doing a string replace for image with video did the trick.

**Comment by dannyv-cloudinary** [comment]
Hi @papa28x4

You can actually specify the resource_type (`image`, `video`, or `raw`) in the optional parameters array when calling the privateDownloadUrl method. Your $downloadUrl should work correctly if you change it to
```
$downloadUrl = (new UploadApi)->privateDownloadUrl("store/tour", "mp4", array(
    "attachment" => true,
    "resource_type" => "video"
));
```

We've listed all the available optional parameters [here in our documentation](https://cloudinary.com/documentation/control_access_to_media#providing_time_limited_access_to_private_media_assets) if you wanted a bit more context around the method.

I hope this helps.

Thanks,
-Danny

**Comment by papa28x4** [comment]
It was because I already did what you just suggested I made the above comment. The base URL doesn't change reflect the resource type

**Comment by papa28x4** [comment]
Hi @dannyv-cloudinary 

```
  public function privateDownloadUrl($publicId, $format, $options = [])
    {
        $params = ApiUtils::finalizeUploadApiParams([
            "public_id"  => $publicId,
            "format"     => $format,
            "type"       => ArrayUtils::get($options, "type"),
            "attachment" => ArrayUtils::get($options, "attachment"),
            "expires_at" => ArrayUtils::get($options, "expires_at"),
        ]);

        ApiUtils::signRequest($params, $this->getCloud());

        return $this->getUploadUrl(AssetType::IMAGE, UploadEndPoint::DOWNLOAD, $params);
    }
```
From my understanding of what I see, the $this->getUploadUrl would always return an image type base url.

**Comment by cloudinaraz** [comment]
Hey @papa28x4 ,

Thank you for brining this to our attention.

This should indeed be fixed so all asset types will be accepted. 
We will make sure to let you know once this is fixed.

**Comment by const-cloudinary** [comment]
Fixed in #286
